src/mps/projmposum.jl

```diff
@@ -128,10 +98,68 @@ ProjMPOSum `P`, and `ortho` is a String which can take
 the values `"left"` or `"right"` depending on the
 sweeping direction of the DMRG calculation.
 """
-function noiseterm(P::ProjMPOSum, phi::ITensor, dir::String)
-  nt = noiseterm(P.pm[1], phi, dir)
-  for n in 2:length(P.pm)
-    nt += noiseterm(P.pm[n], phi, dir)
+function noiseterm(P::AbstractSum, phi::ITensor, dir::String)
+  nt = noiseterm(P.terms[1], phi, dir)
+  for n in 2:length(P.terms)
+    nt += noiseterm(P.terms[n], phi, dir)
   end
   return nt
 end
+
+#
+# Definition of concrete, generic ProjSum type
+#
+
+mutable struct ProjSum{T} <: AbstractSum
+  terms::Vector{T}
```

Comment:
I definitely prefer the field name `terms` here but I worry this will break code.

Also I would remove `mutable` since there is not reason for it since `Vector` is already mutable, but I also worry someone may be relying on that.

Maybe we could just go ahead and make these changes and assume it is only advanced users that might rely on these.

Finally, what about `SerialSum` in contrast with `ParallelSum` that I will define in ITensorParallel.jl?

Reply:
(Maybe `SequentialSum` since `Folds.jl` uses `SequentialEx`.)

Reply:
While we're at it, I think we should define generic field getter and setter functions:
```julia
terms(sum::AbstractSum) = sum.terms
set_terms(sum::AbstractSum, terms) = typeof(sum)(terms)
```
and replace the use of `P.terms` with `terms(P)` in the code in this file.

That will help with code I'm writing in `ITensorParallel.jl` for a generic `ParallelSum` type.

Reply:
We can also be on the safer side and make `set_terms` part of the interface that needs to be overloaded:
```julia
set_terms(sum::AbstractSum, terms) = error("Please implement `set_terms` for the `AbstractSum` type `$(typeof(sum))`.")
```
since the definition above won't work for `AbstractSum` types with other fields (like the Folds executor object that I plan to add to `ITensorParallel.ParallelSum`).